packages/ui/src/components/CardRepeater/CardRepeater.stories.tsx

```diff
@@ -0,0 +1,223 @@
+    event.preventDefault();
+    const item = {
+      id: faker.string.uuid(),
+      value,
+    };
+    action("add")(item);
+    append(item);
+    setValue("");
+    setIsOpen(false);
+  };
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    setValue(event.currentTarget.value);
+  };
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <CardRepeater.Add />
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>Add an item</Dialog.Header>
+        <Dialog.Body>
+          <form onSubmit={add}>
+            <label htmlFor="value">Value</label>
+            <input
+              id="value"
+              type="text"
+              name="value"
+              value={value}
+              onChange={handleChange}
+              required
+            />
+            <Dialog.Footer>
+              <Button type="submit">Add</Button>
+            </Dialog.Footer>
+          </form>
+        </Dialog.Body>
+      </Dialog.Content>
+    </Dialog.Root>
+  );
+};
+
+const themes = ["light", "dark"];
+
+const Template: StoryFn<typeof CardRepeater.Root<CardItem>> = (args) => {
+  const { items: itemsArg } = args;
+  const [items, setItems] = React.useState<CardItem[]>(itemsArg ?? []);
+
+  const handleUpdate = (newItems: CardItem[]) => {
+    action("update")(newItems);
+    setItems(newItems);
+  };
+
+  return (
+    <div
+      data-h2-display="base(grid)"
+      data-h2-grid-template-columns="base(100%) l-tablet(50% 50%)"
+    >
+      {themes.map((theme) => (
+        <div data-h2={theme} key={theme}>
+          <div data-h2-background="base(background)" data-h2-padding="base(x2)">
+            <CardRepeater.Root<CardItem>
+              {...args}
+              items={items}
+              onUpdate={handleUpdate}
+              add={<AddDialog />}
+            >
+              {items.map((item, index) => (
+                <CardRepeater.Card
+                  key={item.id}
+                  index={index}
+                  edit={<EditDialog item={item} index={index} />}
+                >
+                  {item.value}
+                </CardRepeater.Card>
+              ))}
+            </CardRepeater.Root>
+          </div>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const defaultItems = [
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+  {
+    id: faker.string.uuid(),
+    value: faker.company.name(),
+  },
+];
+
+export const WithItems = Template.bind({});
+WithItems.args = {
+  items: defaultItems,
+};
+
+export const MaxItems = Template.bind({});
+MaxItems.args = {
+  ...WithItems.args,
+  max: 4,
+};
```

Comment:
Might be best to set the max to 4 since you are mapping 6 items so you don't have a story that should be impossible. 
Also, not sure if this is a problem but if you are over limit you can start adding again. It only prevents when you are at the limit. Not sure if it is a problem in that it _should_ be impossible to have happen. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/3caad223-ca7c-48f1-9750-47a2fbd8e9f3)

Reply:
haha oops! I added two more for the locked story and forgot to update this. Thanks for catching that 😆 [fix max items story](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9053/commits/1b84350e1a0f3394632fce5fb9d041645055c181).

Also, should be hard to get over the max but I think it makes sense to be safe so I updated it to be `>=` instead of just `===` in [disabled add button is equal to or grater than max](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9053/commits/993ca973314b791a379237f27f467576b8d3d0fc)